src/extractScripts.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+const fs = require('fs');
+const htmlparser2 = require('htmlparser2');
+const tmp = require('tmp');
+
+const SCRIPT = 'script';
+// IMPORTANT: if file naming convention changes,
+// update corresponding strings in testExtractScripts.js
+const TEMP_DIR = 'tempJsScripts';
+
+/**
+ * extractScriptsFromHTML extracts the javascript from within <script>
+ * tags in the HTML test provided and writes each script to a .js file.
+ * It creates a directory "tempJsScripts-{tmp numbers}" in the current
+ * working directory, then puts the .js scripts extracted in the new dir,
+ * named script{number}_{test filename}.js
+ * @param {string} filePath - path to the HTML test file to extract
+ * javascript from within <script> tags.
+ * @param {tmp object} [tempDir=null] - OPTIONAL directory to put
+ * new .js files in. If null, will create one.
+ * @returns {[string]} - the full path of the files that were written.
+ */
+function extractScriptsFromHTML(filePath, tempDir=null) {
+  let scriptCount = 0;
+  let inScript = false;
+  let currentFile = '';
+  const files = [];
+  // Creates a temporary directory to hold all of
+  // the JS scripts in this HTML test file.
+  if (!tempDir) {
+    tempDir = tmp.dirSync({prefix: TEMP_DIR, tmpdir: '.'});
+  }
+
+  const parser = new htmlparser2.Parser(
+      {
+        onopentag(tagname, attribs) {
+          if (tagname === SCRIPT) {
+            inScript = true;
+            const newFileName = '/script' +
```

Comment:
We discussed, it seems since this project will run a few times to migrate all the tests and not be run continuously after that, it's okay to assume we're running on Linux for now and if the need arises we will add platform specific separators.